addons/pos_self_order/static/src/kiosk/components/language_popup/language_popup.js

```diff
@@ -0,0 +1,31 @@
+/** @odoo-module */
+
+import { Component } from "@odoo/owl";
+import { useselfOrder } from "@pos_self_order/kiosk/self_order_kiosk_service";
+import { useService } from "@web/core/utils/hooks";
+
+export class LanguagePopup extends Component {
+    static template = "pos_self_order.LanguagePopup";
+
+    setup() {
+        this.selfOrder = useselfOrder();
+        this.cookie = useService("cookie");
+    }
+
+    get languages() {
+        return this.selfOrder.kiosk_available_languages;
+    }
+
+    get currentLanguage() {
+        return this.selfOrder.currentLanguage;
+    }
+
+    onClickLanguage(language) {
+        this.cookie.setCookie("frontend_lang", language.code);
+        window.location.reload();
+    }
+
+    confirm() {
+        this.props.close();
```

Comment:
As there is a reload when choosing the language, I'm not sure this is useful as the code will never pass into this